### ISSUE TITLE
Bug: Retainer inventories not accurate

### ISSUE THREAD

**Issue by DingFandango** [ffxiv-teamcraft/ffxiv-teamcraft#1243]
**Describe the bug**
When scanning inventory, the lists keep saying certain retainers have an item when the opposite is true. When searching for the item ingame, the item is not in any other retainer but the initial one I find it in.

**To Reproduce**
So far this has been a consistent thing with every list I make. I have reset inventories numerous times and eventually it starts mislabeling which retainers hold what items again.

**Screenshots**
![ffxiv-teamcraft_2019-11-22_09-41-48](https://user-images.githubusercontent.com/23725766/69448302-14a03100-0d0d-11ea-8121-95173d3f87b1.png)
![ffxiv_dx11_2019-11-22_09-48-08](https://user-images.githubusercontent.com/23725766/69448360-37324a00-0d0d-11ea-9d0b-c016411ae435.png)

Notice how the 4 HQ under C identified by Teamcraft is not there in the actual item search but A and B are accurate.

![ffxiv-teamcraft_2019-11-22_09-52-51](https://user-images.githubusercontent.com/23725766/69448663-de16e600-0d0d-11ea-90e2-d4aae7fb7647.png)
![ffxiv_dx11_2019-11-22_09-53-19](https://user-images.githubusercontent.com/23725766/69448691-ee2ec580-0d0d-11ea-97db-3fdfc997014e.png)

**Desktop (please complete the following information):**
 - Windows 10
 - Chrome
 - Version 6.26

**Comment by Supamiu** [comment]
Teamcraft doesn't read your retainer's inventory, it reads the packets that are updating it.

In order to find what's going on, I need specific steps to find out when the inventory starts to be wrong, as it's the reason why this bug exists.

Please add steps to reproduce the unsync state in order to make me able to fix it. I know it's not easy, and i'm trying to reproduce it too, but for now, I couldn't and it's why I need more people to try and reproduce it.

**Comment by DingFandango** [comment]
I reset my inventory again and will watch teamcraft closely to see what is causing the issue. So far after resetting my inventory and going through every retainer there are no mislabeling. I'll do some crafting and see if that causes the issue which was when I noticed the problem the first time around.

**Comment by DingFandango** [comment]
Alright the incorrect listings seems to be triggered now. What happened today before it happened was:

Duty Roulettes
Automatic AFK timer activating

Gathering, crafting, attaching materia and other minor things did not seem to trigger this, but I have not done  roulettes for a couple days and this is the first time I have gone on the automatic afk today. As of now, the autofill works, yet Teamcraft is not saying I have items in my inventory that I just withdrew.

![ffxiv-teamcraft_2019-11-23_21-42-22](https://user-images.githubusercontent.com/23725766/69490283-2501fa00-0e3a-11ea-9d6f-451037d8b6f5.png)
![ffxiv_dx11_2019-11-23_21-42-57](https://user-images.githubusercontent.com/23725766/69490288-3a772400-0e3a-11ea-8ec5-9a1569b3ede2.png)

**Comment by Supamiu** [comment]
How did you move the items? Drag and drop? Right click? Every single small bit of detail is important :)

**Comment by DingFandango** [comment]
I moved the items by right clicking and "Retrieve Quantity" as well as "Retrieve from Retainer". I deposited them by "Move to Retainer", too.

**Comment by DingFandango** [comment]
I noticed today that after resetting my inventory again, Teamcraft only initially lists my chocobo saddlebag instead of both my inventory and chocobo saddlebag. Also before resetting my inventory today, lists would not show changes in my inventory even with autofill on but still would update the list. If I pulled out a green glider leather for example, the list would still mark it as complete, but it would not say that I have 1 green glider leather in my inventory.

![ffxiv-teamcraft_2019-11-25_11-45-20](https://user-images.githubusercontent.com/23725766/69572480-12daa580-0f79-11ea-9d5d-616798c347da.png)

**Comment by Jyntax** [comment]
I had the same issues. For me it seems to happen every now and then after an update of the app.

Fixed it every time by downloading the last version before the update, installing it and testing if everything works again - if yes I updated the client and it worked fine then.

Don't know tho why the issue even happens on my end since it's always after an update as it seems.

**Comment by ProbablyAimee** [comment]
I can confirm that this happened for me after the last two Teamcraft updates. I run the update as admin before starting FFXIV but inventory items appear listed on retainers who never had them. Clearing the inventory and reloading fixes that, but then a PC restart is required or Teamcraft autofill on lists will be very slow and freeze the desktop app.

**Comment by DingFandango** [comment]
I think I've figured out why it's saying my retainers have items they don't have. I inspected the inventories section and the retainers who have incorrect listings show that they have ALL the items in my own inventory, armory and other retainers' inventory for some reason. 

![image](https://user-images.githubusercontent.com/23725766/71373112-f4150200-256a-11ea-9d77-2bbfea80b42b.png)

For example in this picture it's showing items that are in my armory and yet my retainers have them.
